### PR TITLE
Release cleanup

### PR DESCRIPTION
**Pull request by aryszka**
one (biggest) step of a refactoring. not making a pr directly to master, but only into the review branch of release-cleanup-review. the whole refactoring is not yet done, but it is worth reviewing in separate steps. in this step i removed the common, empty interface package.

### CODE REVIEW

File: filters/filters.go [hunk 1]
```diff
@@ -1,38 +1,59 @@
...
+// Typically, there is a single Spec instance of each implementation in a running process, which can create multiple filter
+// instances with different config defined in the configuration on every update.
+type Spec interface {
+
+	// The name of the Spec is used to identify in the configuration which spec a filter is based on.
+	Name() string
+
+	// When the program settings are updated, and they contain filters based on a spec, MakeFilter is
+	// called, and the filter id and the filter specific settings are provided. Returns a filter.
+	CreateFilter(config []interface{}) (Filter, error)
```
Comment (lmineiro) [inline filters/filters.go]: nitpicking here, but I really like the "New...()" naming convention. It seems more idiomatic to me

Comment (danpersa) [inline filters/filters.go]: In the comment is specified MakeFilter, but the method is called CreateFilter

File: filters/headerfilter.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+func (spec *headerFilter) CreateFilter(config []interface{}) (Filter, error) {
+	key, value, err := headerFilterConfig(config)
+	return &headerFilter{typ: spec.typ, key: key, value: value}, err
+}
+
+func (f *headerFilter) Request(ctx FilterContext) {
+	if f.typ == requestHeader {
+		req := ctx.Request()
+		if f.key == "Host" {
```
Comment (lmineiro) [inline filters/headerfilter.go]: Should it be case insensitive? http://tools.ietf.org/html/rfc7230#section-3.2 already states they should. I wouldn't expect the filter to force it

File: filters/redirect.go [hunk 3]
```diff
@@ -16,21 +15,21 @@ type Redirect struct {
...
-	if len(c) != 2 {
+	if len(config) != 2 {
 		return invalidArgs()
 	}
 
-	code, ok := c[0].(float64)
+	code, ok := config[0].(float64)
 	if !ok {
 		return invalidArgs()
 	}
```
Comment (lmineiro) [inline filters/redirect.go]: Should we test the status code range and fail if not ok?

File: filters/filters.go (hunk 1 above)
Comment (aryszka) [inline filters/filters.go]: yup:
New: i agree and it is in the pipe to fix it globally, as part of this same refactor. this pull request is only about the removal of the empty interface package.
Comment: it is also the part of this refactor, to go through the whole documentation, and fix/extend/update it. As reviewers, do you prefer to keep the doc continuously in sync?

File: filters/headerfilter.go (hunk 2 above)
Comment (aryszka) [inline filters/headerfilter.go]: Right! Fixing it.

File: filters/redirect.go (hunk 3 above)
Comment (aryszka) [inline filters/redirect.go]: i think in this case we can save that code and leave to the audience whether they want to do something irregular.

File: filters/filters.go (hunk 1 above)
Comment (lmineiro) [inline filters/filters.go]: This being a PR to another branch that will get reviewed I believe it should be fine. The final PR for the destination branch should contain the docs updates

**Pull request merged**